test/test_getsetall.jl

```diff
@@ -0,0 +1,21 @@
+module TestGetSetAll
+using Test
+using Accessors
+
+@testset "getall" begin
+    obj = (a=1, b=2.0, c='3')
+    @test @inferred(getall(obj, @optic(_.a))) === (1,)
+    @test @inferred(getall(obj, @optic(_ |> Properties()))) === (1, 2.0, '3')
+    @test @inferred(getall(obj, @optic(_ |> Elements()))) === (1, 2.0, '3')
+    
+    obj = (a=1, b=((c=3, d=4), (c=5, d=6)))
+    @test @inferred(getall(obj, @optic(_.b |> Elements() |> _.c))) === (3, 5)
+    @test @inferred(getall(obj, @optic(_.b |> Elements() |> Properties()))) === (3, 4, 5, 6)
+    @test getall(obj, @optic(_.b |> Elements() |> Properties() |> _ * 3)) === (9, 12, 15, 18)
+    @test_broken (@inferred(getall(obj, @optic(_.b |> Elements() |> Properties() |> _ * 3))); true)
```

Comment:
Can you add a few more tests for when inference works and when it is broken? In particular, I am curious if it is a problem that we have a function lens here. As I understand the compiler has some heuristics not to specialize too much on arguments that are functions.